include/seqan3/alignment/matrix/detail/score_matrix_single_column.hpp

```diff
@@ -0,0 +1,254 @@
+ * seqan3::detail::affine_cell_proxy to simplify the access to the correct values without knowing the internal
+ * tuple layout returned by the seqan3::views::zip view.
+ */
+template <typename score_t>
+class score_matrix_single_column<score_t>::matrix_iterator
+{
+private:
+
+    //!\brief The type of the zipped score column.
+    using matrix_column_t = decltype(views::zip(std::declval<physical_column_t &>(),
+                                                std::declval<physical_column_t &>(),
+                                                std::declval<virtual_column_t &>()));
+
+    //!\brief The transform adaptor to convert the tuple from the zip view into a seqan3::detail::affine_cell_type.
+    static constexpr auto transform_to_affine_cell = std::views::transform([] (auto && tpl)
+    {
+        using fwd_tuple_t = decltype(tpl);
+        return affine_cell_proxy<remove_cvref_t<fwd_tuple_t>>{std::forward<fwd_tuple_t>(tpl)};
+    });
+
+    //!\brief The pointer to the underlying matrix.
+    score_matrix_single_column * m_host_ptr{nullptr};
+    //!\brief The current column index.
+    size_t m_current_column{};
+    //!\brief The column index of the column behind the last one.
+    size_t m_end_column{};
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief The value type.
+    using value_type = decltype(std::declval<matrix_column_t>() | transform_to_affine_cell);
+    //!\brief The reference type.
+    using reference = value_type;
+    //!\brief The pointer type.
+    using pointer = void;
+    //!\brief The difference type.
+    using difference_type = std::ptrdiff_t;
+    //!\brief The iterator category.
+    using iterator_category = std::input_iterator_tag;
+    //!\}
+
+    /*!\name Constructor, assignment and destructor
+     * \{
+     */
+    matrix_iterator() noexcept = default; //!< Defaulted.
+    matrix_iterator(matrix_iterator const &) noexcept = default; //!< Defaulted.
+    matrix_iterator(matrix_iterator &&) noexcept = default; //!< Defaulted.
+    matrix_iterator & operator=(matrix_iterator const &) noexcept = default; //!< Defaulted.
+    matrix_iterator & operator=(matrix_iterator &&) noexcept = default; //!< Defaulted.
+    ~matrix_iterator() = default; //!< Defaulted.
+
+    /*!\brief Initialises the iterator from the underlying matrix.
+     *
+     * \param[in] host_matrix The underlying matrix.
+     */
+    explicit matrix_iterator(score_matrix_single_column & host_matrix) noexcept :
+        m_host_ptr{std::addressof(host_matrix)},
+        m_end_column{m_host_ptr->m_number_of_columns}
+    {}
+    //!\}
+
+    /*!\name Access operators
+     * \{
+     */
+    //!\brief Returns the range over the current column.
+    reference operator*() const
+    {
+        return views::zip(m_host_ptr->m_optimal_column, m_host_ptr->m_horizontal_column, m_host_ptr->m_vertical_column)
+             | transform_to_affine_cell;
```

Comment:
```suggestion
               | transform_to_affine_cell;
```

Reply:
I think it is aligned with the guide about piping views such that all views start on the same column.